### PR TITLE
Rename rebalancer to liquidity manager

### PR DESCRIPTION
**Pull request by amirylm**
## Motivation

Renaming packages, internal structs and contracts from `Rebalancer` to `Liquidity Manager`.

## Solution

Align plugin and contracts accordingly.

### CODE REVIEW

File: core/services/ocr2/plugins/liquiditymanager/chain/evm/liquidity_manager.go [hunk 1]
```diff
@@ -1,18 +1,18 @@
...
+	"github.com/smartcontractkit/chainlink/v2/core/services/ocr2/plugins/liquiditymanager/models"
 )
 
-// Rebalancer is an abstraction of the rebalancer contract.
+// LiquidityManager is an abstraction of the rebalancer contract.
 //
-//go:generate mockery --quiet --name Rebalancer --output ./mocks --filename rebalancer_mock.go --case=underscore
-type Rebalancer interface {
+//go:generate mockery --quiet --name LiquidityManager --output ./mocks --filename liquidity_manager_mock.go --case=underscore
+type LiquidityManager interface {
```
Comment (RyanRHall) [inline core/services/ocr2/plugins/liquiditymanager/chain/evm/liquidity_manager.go:16]: should this be `EVMLiquidityManager`?

**Review by RyanRHall — COMMENTED** [review]
Only question here is the separation of `liquidiymanager` and `evmliquiditymanager` - should these be combined into one? are we expecting non-evm liquidity managers will use the chain-agnostic `liquiditymanager` interfaces?

File: core/services/ocr2/plugins/liquiditymanager/chain/evm/liquidity_manager.go (hunk 1 above)
Comment (amirylm) [reply core/services/ocr2/plugins/liquiditymanager/chain/evm/liquidity_manager.go:16]: Good point, the interface itself is not chain specific and therefore can be extracted from this package

File: contracts/src/v0.8/liquiditymanager/test/helpers/RebalancerHelper.sol
Comment (makramkd) [inline contracts/src/v0.8/liquiditymanager/test/helpers/RebalancerHelper.sol:1]: Rename this to LiquidityManagerHelper?

File: core/gethwrappers/liquiditymanager/generated/liquiditymanager/liquiditymanager.go [hunk 2]
```diff
@@ -30,209 +30,209 @@ var (
...
...766137de565b82358281111561418557600080fd5b8301601f8101881361419657600080fd5b80356141a46138b782613850565b81815260059190911b8201860190868101908a8311156141c357600080fd5b8784015b8381101561426f578035878111156141de57600080fd5b85016080818e037fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffe001121561421257600080fd5b61421a6137b5565b8a820135815260408201358b82015261423560608301613622565b604082015260808201358981111561424d5760008081fd5b61425b8f8d83860101613961565b6060830152508452509188019188016141c7565b508452505050828401358281111561428657600080fd5b61429288828601614027565b948201949094529695505050505050565b600073ffffffffffffffffffffffffffffffffffffffff8088168352808716602084015280861660408401525083606083015260a0608083015261337360a083018461353f565b6000602082840312156142fc57600080fd5b815167ffffffffffffffff81111561431357600080fd5b8201601f8101841361432457600080fd5b80516143326138b78261391b565b81815285602083850101111561434757600080fd5b613f6482602083016020860161351b565b73ffffffffffffffffffffffffffffffffffffffff8516815283602082015260806040820152600061438d608083018561353f565b8281036060840152613373818561353f565b600082516143b181846020870161351b565b919091019291505056fea164736f6c6343000813000a",
```
Comment (makramkd) [inline core/gethwrappers/liquiditymanager/generated/liquiditymanager/liquiditymanager.go:50]: The only diff in this bytecode is the new `typeAndVersion` constant.

File: core/services/ocr2/plugins/liquiditymanager/bridge/arb/l1_to_l2.go [hunk 3]
```diff
@@ -101,7 +101,7 @@ func NewL1ToL2Bridge(
 	}
 
 	// figure out which gateway to watch for the token on L2
-	l1Rebalancer, err := rebalancer.NewRebalancer(l1RebalancerAddress, l1Client)
+	l1Rebalancer, err := liquiditymanager.NewLiquidityManager(l1RebalancerAddress, l1Client)
```
Comment (makramkd) [inline core/services/ocr2/plugins/liquiditymanager/bridge/arb/l1_to_l2.go]:
```suggestion
	l1LiquidityManager, err := liquiditymanager.NewLiquidityManager(l1RebalancerAddress, l1Client)
```

File: core/services/ocr2/plugins/liquiditymanager/bridge/arb/l1_to_l2.go
Comment (makramkd) [inline core/services/ocr2/plugins/liquiditymanager/bridge/arb/l1_to_l2.go:1]: Still some stragglers in this file, can find/replace 'rebalancer' to change them

File: core/services/ocr2/plugins/liquiditymanager/bridge/arb/l2_to_l1.go [hunk 4]
```diff
@@ -108,7 +108,7 @@ func NewL2ToL1Bridge(
 		return nil, fmt.Errorf("register filter for Arbitrum L1 to L2 bridge: %w", err)
 	}
 
-	l1Rebalancer, err := rebalancer.NewRebalancer(l1RebalancerAddress, l1Client)
+	l1Rebalancer, err := liquiditymanager.NewLiquidityManager(l1RebalancerAddress, l1Client)
```
Comment (makramkd) [inline core/services/ocr2/plugins/liquiditymanager/bridge/arb/l2_to_l1.go]:
```suggestion
	l1LiquidityManager, err := liquiditymanager.NewLiquidityManager(l1LiquidityManagerAddress, l1Client)
```

File: core/services/ocr2/plugins/liquiditymanager/bridge/testonlybridge/l1_to_l1.go [hunk 5]
```diff
@@ -32,15 +32,15 @@ var (
...
+	FinalizationStepCompletedTopic = liquiditymanager.LiquidityManagerFinalizationStepCompleted{}.Topic()
 )
 
 type testBridge struct {
 	sourceSelector   models.NetworkSelector
 	destSelector     models.NetworkSelector
-	sourceRebalancer rebalancer.RebalancerInterface
-	destRebalancer   rebalancer.RebalancerInterface
+	sourceRebalancer liquiditymanager.LiquidityManagerInterface
+	destRebalancer   liquiditymanager.LiquidityManagerInterface
```
Comment (makramkd) [inline core/services/ocr2/plugins/liquiditymanager/bridge/testonlybridge/l1_to_l1.go]:
```suggestion
	sourceLM liquiditymanager.LiquidityManagerInterface
	destLM   liquiditymanager.LiquidityManagerInterface
```

File: core/services/ocr2/plugins/liquiditymanager/chain/evm/evm.go [hunk 6]
```diff
@@ -11,15 +11,15 @@ import (
...
+	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/liquiditymanager/generated/liquiditymanager"
 	"github.com/smartcontractkit/chainlink/v2/core/logger"
-	"github.com/smartcontractkit/chainlink/v2/core/services/ocr2/plugins/rebalancer/models"
+	"github.com/smartcontractkit/chainlink/v2/core/services/ocr2/plugins/liquiditymanager/models"
 )
 
-var _ Rebalancer = &EvmRebalancer{}
+var _ LiquidityManager = &EvmRebalancer{}
 
 type EvmRebalancer struct {
```
Comment (makramkd) [inline core/services/ocr2/plugins/liquiditymanager/chain/evm/evm.go]:
```suggestion
type EVMLM struct {
```

File: core/services/ocr2/plugins/liquiditymanager/internal/rebalancer_integration_test.go [hunk 7]
```diff
@@ -690,9 +690,9 @@ func deployContracts(
 
 		// deploy the rebalancer and set the liquidity container to be the lock release pool
 		ch := mustGetChainByEvmID(t, chainID)
-		rebalancerAddr, _, _, err := rebalancer.DeployRebalancer(owner, backend, wethAddress, ch.Selector, lockReleasePoolAddress, big.NewInt(0))
+		rebalancerAddr, _, _, err := liquiditymanager.DeployLiquidityManager(owner, backend, wethAddress, ch.Selector, lockReleasePoolAddress, big.NewInt(0))
 		require.NoError(t, err, "failed to deploy Rebalancer contract")
```
Comment (makramkd) [inline core/services/ocr2/plugins/liquiditymanager/internal/rebalancer_integration_test.go]: Still some stragglers here, find/replace Rebalancer should catch them

File: core/services/ocr2/validate/validate.go [hunk 8]
```diff
@@ -22,10 +22,10 @@ import (
 	"github.com/smartcontractkit/chainlink/v2/core/services/job"
 	"github.com/smartcontractkit/chainlink/v2/core/services/ocr2/plugins/ccip/config"
 	dkgconfig "github.com/smartcontractkit/chainlink/v2/core/services/ocr2/plugins/dkg/config"
+	rebalancermodels "github.com/smartcontractkit/chainlink/v2/core/services/ocr2/plugins/liquiditymanager/models"
```
Comment (makramkd) [inline core/services/ocr2/validate/validate.go]:
```suggestion
	lmmodels "github.com/smartcontractkit/chainlink/v2/core/services/ocr2/plugins/liquiditymanager/models"
```

File: core/scripts/ccip/liquiditymanager/opstack/prove_withdrawal.go [hunk 9]
```diff
@@ -84,7 +84,7 @@ func ProveWithdrawalViaRebalancer(
 
 	encodedPayload := proveMessagePayload(l1Client, l2Client, l2TxHash, optimismPortalAddress, l2OutputOracleAddress)
 
-	l1Rebalancer, err := rebalancer.NewRebalancer(l1RebalancerAddress, l1Client)
+	l1Rebalancer, err := liquiditymanager.NewLiquidityManager(l1RebalancerAddress, l1Client)
```
Comment (makramkd) [inline core/scripts/ccip/liquiditymanager/opstack/prove_withdrawal.go:87]:
```suggestion
	l1LiquidityManager, err := liquiditymanager.NewLiquidityManager(l1RebalancerAddress, l1Client)
```

**Review by makramkd — APPROVED** [review]

**Review by anirudhwarrier — APPROVED** [review]
LG, tested the setup

**Review by chainchad — APPROVED** [review]
Approving for CI changes.

**Pull request merged**